### ISSUE TITLE
Incorrect visibility error when setter precedes getter

### ISSUE THREAD

**Issue by zfeed** [microsoft/TypeScript#49685]
# Bug Report

### 🔎 Search Terms

private/public class property order

### 🕗 Version & Regression Information

- This is the behavior in every version from 4.3.5 to 4.7.4

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.7.4#code/PTAEFEA8EMFsAcA2BTUB3AlgFwBYHsBXLUZAJ1L1IFgAoAY0WgGcnQBVJs0Ab1tFADEAOzjIAhAC5QTLKQxCA5gG5afUPAIAjRBjqgFyYiNjIAFAEoea-qUMFSQ0LgxMAdMNEqa-AL6rv6nIAbtBYqJxGoqbGyFIycoqWvAH8zm4eJqAAvKAxXr60fjS0dHhCMqAEnKTZucho7NUWXrQguXgk5JS0VWSuef6tYFBwSKiYuJ0U1PSMLI1k4F01yfwZ4nGy8spqavDBoeGGuVExmwkKSdZOOC7uMbUDAUV7Wjp6BpEmFlYpoLZYeyONL3TxqIovehlCq9UhLaaPeoLOHLZr+NpkaY9arwyj9ME0IA)

### 💻 Code

```ts
// Example without error
class User {
  #name!: string;

  public get name() {
    return this.#name;
  }

  private set name(name: string) {
    this.#name = name;
  }
}

const user = new User();

// no error
user.name;


// Example with error
class UserError {
  #name!: string;
  
  private set name(name: string) {
    this.#name = name;
  }

  public get name() {
    return this.#name;
  }
}

const userError = new UserError();

// error: Property 'name' is private and only accessible within class 'UserError'.
userError.name;

```

### 🙁 Actual behavior

If we firstly define a private setter for a property and then a public getter for the same property we receive an error when accessing the getter - `Property is private and only accessible within class`. If we change the order the error doesn't appear

### 🙂 Expected behavior
Any order is acceptable and doesn't produce an error
